Modules/ApplicationInsights/2.0/deploy.json

```diff
@@ -0,0 +1,199 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "appInsightsName": {
+      "type": "string",
+      "metadata": {
+        "description": "Required. Name of the Application Insights"
+      }
+    },
+    "appInsightsType": {
+      "type": "string",
+      "allowedValues": [
+        "web",
+        "java",
+        "other"
+      ],
+      "defaultValue": "web",
+      "metadata": {
+        "description": "Optional. Application type"
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Optional. Location for all Resources"
+      }
+    },
+    "storageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "Required. Storage Account Name"
+      }
+    },
+    "storageAccountsku": {
+      "type": "string",
+      "allowedValues": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_RAGRS",
+        "Standard_ZRS",
+        "Premium_LRS",
+        "Premium_ZRS",
+        "Standard_GZRS",
+        "Standard_RAGZRS"
+      ],
+      "defaultValue": "Standard_GRS",
+      "metadata": {
+        "description": "Optional. Storage Account sku type"
+      }
+    },
+    "cuaId": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered"
+      }
+    },
+    "tagEnvironment": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The name of the Environment"
+      }
+    },
+    "tagProject": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The name of the project"
+      }
+    },
+    "tagApplication": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The name of the application"
+      }
+    },
+    "tagOwner": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The business owner for the application"
+      }
+    },
+    "tagOwnerEmail": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The Email address of the business owner for the application"
+      }
+    }
+  },
+  "variables": {
+    "appInsightsName": "[parameters('appInsightsName')]",
+    "appInsightsresourceId": "[resourceId('Microsoft.Insights/components',parameters('appInsightsName'))]",
+    "apiVersion": "[providers('Microsoft.Insights','components').apiVersions[0]]",
+    "storageAccountName": "[parameters('storageAccountName')]",
+    "pidName": "[concat('pid-',parameters('cuaId'))]"
+  },
+  "resources": [
+    {
+      "condition": "[not(empty(parameters('cuaId')))]",
```

Comment:
@bennage do we want to include this value in every single template? I think is a good idea if and only if we add a condition to prevent the deployment of this resource. This resource is related to the Partner Tracking ACR ID.

Reply:
In every module